codegen/src/lib.rs

```diff
@@ -8,6 +8,20 @@ use quote::quote;
 use syn::Error;
 use syn::ItemFn;
 
+//We can't use procedural macros as expressions due to rust limitations
+//so this generates a function instead
+#[proc_macro]
+pub fn make_env(input: TokenStream) -> TokenStream {
+    let env_arg = syn::parse_macro_input!(input as syn::LitStr);
+    let app_size_str = std::env::var(env_arg.value()).unwrap();
+    let size = app_size_str.parse::<usize>().unwrap();
+
+    let result = quote! {
+        fn do_parse_env() -> usize { #size }
```

Comment:
Could this macro directly return `#size`?

Reply:
You mean the make_env macro? I don't think so. Procedural macros can't generate expressions so we have to indirectly call a function. The do_parse_env function does get inlined.

Reply:
Ah, right. But the function could be named `read_#env_var_name`. Actually, it's not parsing but reading what has been parsed by the macro.